plugins/action/zos_job_submit.py

```diff
@@ -148,16 +154,20 @@ def run(self, tmp=None, task_vars=None):
                     src=tmp_src,
                     dest=dest_path,
                     mode="0600",
-                    _original_basename=source_rel,
-                )
-            )
-            result.update(
-                self._execute_module(
-                    module_name="copy",
-                    module_args=copy_module_args,
-                    task_vars=task_vars,
+                    force=True,
+                    remote_src=True,
                 )
             )
+            copy_task = copy.deepcopy(self._task)
+            copy_task.args = copy_module_args
+            zos_copy_action_module = ZosCopyActionModule(task=copy_task,
+                                                         connection=self._connection,
+                                                         play_context=self._play_context,
+                                                         loader=self._loader,
+                                                         templar=self._templar,
+                                                         shared_loader_obj=self._shared_loader_obj)
+            result.update(zos_copy_action_module.run(task_vars=task_vars))
+            module_args["src"] = dest_path
             result.update(
```

Comment:
Please add a validation when zos_copy action fails, this will avoid some headaches when debugging. Take a look into zos_unarchive, the code checks result.get('msg') I believe, is enough just adding a message that copying to remote failed.